clippy_lints/src/methods/map_flatten.rs

```diff
@@ -39,7 +41,8 @@ pub(super) fn check<'tcx>(
             // `(...).map(...)` has type `impl Iterator<Item=impl Iterator<...>>
             "flat_map"
         };
-        let func_snippet = snippet(cx, map_arg.span, "..");
+        let mut applicability = Applicability::MachineApplicable;
+        let func_snippet = snippet_with_applicability(cx, map_arg.span, "..", &mut applicability);
         let hint = format!(".{0}({1})", method_to_use, func_snippet);
         span_lint_and_sugg(
```

Comment:
This lint emission of `MAP_FLATTEN` should also be affected by your changes. It would probably be good to have a function that selects the message style and emits the lint, so it can be used for iterators and `Result`/`Option` types.

From your comment, it looks like this is still on the TODO list, this is therefore just a reminder for me :upside_down_face: 

BTW nice usage of `snippet_with_applicability` :+1:

Reply:
It's finished now 😃, feel free to give any kind of feedback when you got time~